From: =?utf-8?q?R=C3=A9mi_Vanicat?= <vanicat@debian.org>
Date: Sun, 11 Nov 2018 14:14:06 +0100
Subject: set tramp-histfile-override in tramp test

In Debian building system, we can't wrote in home direcetory
---
 t/magit-tests.el | 27 ++++++++++++++-------------
 1 file changed, 14 insertions(+), 13 deletions(-)

diff --git a/t/magit-tests.el b/t/magit-tests.el
index 523cc51..d414333 100644
--- a/t/magit-tests.el
+++ b/t/magit-tests.el
@@ -83,19 +83,20 @@
                       shell-file-name)))
              ((cdr (assq 'tramp-login-args sudo-method)) nil))
     (magit-with-test-directory
-     (setq default-directory
-           (concat (format "/sudo:%s@localhost:" (user-login-name))
-                   default-directory))
-     (magit-test-init-repo "repo")
-     (magit-test-magit-toplevel)
-     (should (equal (magit-toplevel   "repo/.git/")
-                    (expand-file-name "repo/")))
-     (should (equal (magit-toplevel   "repo/.git/objects/")
-                    (expand-file-name "repo/")))
-     (should (equal (magit-toplevel   "repo-link/.git/")
-                    (expand-file-name "repo-link/")))
-     (should (equal (magit-toplevel   "repo-link/.git/objects/")
-                    (expand-file-name "repo/"))))))
+     (let ((tramp-histfile-override default-directory))
+       (setq default-directory
+             (concat (format "/sudo:%s@localhost:" (user-login-name))
+                     default-directory))
+       (magit-test-init-repo "repo")
+       (magit-test-magit-toplevel)
+       (should (equal (magit-toplevel   "repo/.git/")
+                      (expand-file-name "repo/")))
+       (should (equal (magit-toplevel   "repo/.git/objects/")
+                      (expand-file-name "repo/")))
+       (should (equal (magit-toplevel   "repo-link/.git/")
+                      (expand-file-name "repo-link/")))
+       (should (equal (magit-toplevel   "repo-link/.git/objects/")
+                      (expand-file-name "repo/")))))))
 
 (ert-deftest magit-toplevel:submodule ()
   (let ((find-file-visit-truename nil))
